pkg/controller/container-runtime-config/helpers.go

```diff
@@ -431,14 +435,18 @@ func validateUserContainerRuntimeConfig(cfg *mcfgv1.ContainerRuntimeConfig) erro
 	}
 
 	ctrcfg := cfg.Spec.ContainerRuntimeConfig
-	if ctrcfg.PidsLimit > 0 && ctrcfg.PidsLimit < minPidsLimit {
-		return fmt.Errorf("invalid PidsLimit %q, cannot be less than 20", ctrcfg.PidsLimit)
+	if ctrcfg.PidsLimit != nil && *ctrcfg.PidsLimit > 0 && *ctrcfg.PidsLimit < minPidsLimit {
```

Comment:
Why are negative values silently ignored here?

Reply:
This should fail on negative values. @haircommander iirc you were testing out something by setting this to 0 and/or -1, should -1 be a valid value?

Reply:
confusingly, negative values result in the "max" number of allowed, but zero (on my machine) returns `37725`

I'm putting together a cri-o PR to test this

Reply:
https://github.com/cri-o/cri-o/pull/4614

Reply:
> confusingly, negative values result in the "max" number of allowed, but zero (on my machine) returns 37725

Are these features, documented implementation details, or undocumented implementation details?

The bug originally started just with “The MCO should validate configurations and reject invalid ones”; it’s not clear to me that it requests explicitly implementing all the possible obscure values. AFAICT (but I haven’t looked into it deeply), the original reporter might have been content with just rejecting `pidsLimit: 0` as invalid, rather than explicitly supporting that (or negative values) as special cases.

(OTOH, sure, if CRI-O implements `pidsLimit: 0`, it’s plausible that the MCO should allow configuring that. But then again, OpenShift very intentionally does not expose every single feature of the underlying components. I have no idea whether the 0 and negative cases qualify as “useful OpenShift features” or “obscure implementation details OpenShift should hide”.)

Reply:
> Are these features, documented implementation details, or undocumented implementation details?
AFACT, undocumented implementation details. As such, I am leaning towards not allowing them here. 

One day, we may get a request to allow `max` pids_limit, and then we can decide whether it's worth adding the logic. Until then, I don't think it's worth the extra logic to allow 0 or -1

Reply:
@QiWang19 let's not change the existing behavior so we don't accidentally break someones use case. Let's leave 0 as an allowed value as it already was and we can fix the error logged for an invalid value. 
You can now control the pids limit via the kubelet, so we should open up a follow up PR to deprecate it here and document how to use kubelet instead to set it.

Reply:
@umohnani8  Could you elaborate how to leave 0 as allowed value? Current behavior is ignoring the 0 and negative values, no error will be logged and, no new configuration will be rolled out, pidslimit will be the default value 1024. So just remove the changes of the pidslimit in this PR is enough?

Reply:
So throw an error if `pidsLimit < 0 || (pidsLimt > 0 && pidsLimit < minPidsLimit)` This way we cover negative values and any value less than 20 that is not 0. We want 0 to be a valid value as it was.